File: backend/app/schemas/screening_database.py
from datetime import datetime

from pydantic import BaseModel, Field

from app.schemas.common import SourceFileResponse


class ImportJobSummaryResponse(BaseModel):
    import_id: str
    status: str
    file_name: str
    file_type: str
    file_size: int | None = None
    detected_rows: int = 0
    success_rows: int = 0
    failed_rows: int = 0
    validation_error_count: int = 0
    created_at: datetime | None = None
    started_at: datetime | None = None
    finished_at: datetime | None = None
    created_by: str | None = None
    source_set_hash: str | None = None
    error_summary: str | None = None


class ImportJobListResponse(BaseModel):
    imports: list[ImportJobSummaryResponse] = Field(default_factory=list)
    total: int = 0


class ImportJobDetailResponse(ImportJobSummaryResponse):
    source_type: str
    source_file_hash: str | None = None
    source_file_path: str | None = None
    source_file_modified_at: datetime | None = None
    parsed_rows: int = 0
    valid_rows: int = 0
    invalid_rows: int = 0
    warning_rows: int = 0
    merged_rows: int = 0
    skipped_rows: int = 0
    duplicate_identifier_count: int = 0
    source_files: list[SourceFileResponse] = Field(default_factory=list)


class StageUploadResponse(BaseModel):
    status: str  # "staged" | "staged_pdf_needs_review"
    file_name: str
    file_type: str
    file_size: int
    message: str
    next_step: str
    needs_review: bool = False

File: frontend/src/types/screening-database.ts
export type ImportJobSummary = {
  import_id: string;
  status: string;
  file_name: string;
  file_type: string;
  file_size: number | null;
  detected_rows: number;
  success_rows: number;
  failed_rows: number;
  validation_error_count: number;
  created_at: string | null;
  started_at: string | null;
  finished_at: string | null;
  created_by: string | null;
  source_set_hash: string | null;
  error_summary: string | null;
};

export type ImportSourceFile = {
  file_id: string | null;
  file_name: string;
  file_path: string | null;
  file_type: string;
  sha256: string;
  size_bytes: number;
  modified_at: string | null;
  parse_status: string | null;
  row_count: number | null;
  warning_count: number | null;
  error_message: string | null;
  parse_error_summary?: string | null;
  discovered_at?: string | null;
};

export type ImportJobDetail = ImportJobSummary & {
  source_type: string;
  source_file_hash: string | null;
  source_file_path: string | null;
  source_file_modified_at: string | null;
  parsed_rows: number;
  valid_rows: number;
  invalid_rows: number;
  warning_rows: number;
  merged_rows: number;
  skipped_rows: number;
  duplicate_identifier_count: number;
  source_files: ImportSourceFile[];
};

export type ImportJobListResponse = {
  imports: ImportJobSummary[];
  total: number;
};

export type StageUploadResponse = {
  status: string;
  file_name: string;
  file_type: string;
  file_size: number;
  message: string;
  next_step: string;
  needs_review: boolean;
};
